plugins/readtimer.koplugin/main.lua

```diff
@@ -46,9 +60,54 @@ function ReadTimer:init()
             })
         end
     end
+
+    self.additional_header_content_func = function()
+        if self:scheduled() then
+            local hours, minutes, dummy = self:remainingTime(1)
+            local timer_info = string.format("%02d:%02d", hours, minutes)
+            return self.timer_symbol .. timer_info .. " "
+        else
+            return ""
+        end
+    end
+    self.additional_footer_content_func = function()
+        if self:scheduled() then
+            local item_prefix = self.ui.view.footer.settings.item_prefix
+            local hours, minutes, dummy = self:remainingTime(1)
+            local timer_info = string.format("%02d:%02d", hours, minutes)
+
+            if item_prefix == "icons" then
+                return self.timer_symbol .. " " .. timer_info
+            elseif item_prefix == "compact_items" then
+                return self.timer_symbol .. timer_info
+            else
+                return self.timer_letter .. ": " .. timer_info
+            end
+        else
+            return ""
+        end
+    end
+
     self.ui.menu:registerToMainMenu(self)
 end
 
+function ReadTimer:update_status_bars(seconds)
+    if self.readtimer_show_value_in_footerheader then
+        UIManager:broadcastEvent(Event:new("UpdateHeader"))
+    end
+    if self.readtimer_show_value_in_footer then
+        UIManager:broadcastEvent(Event:new("UpdateFooter", true))
+    end
+    -- if seconds schedule 1ms later
+    if seconds and seconds >= 0 then
+        UIManager:scheduleIn(math.max(math.floor(seconds)%60, 0.001), self.update_status_bars, self)
+    elseif seconds and seconds < 0 and self:scheduled() then
+        UIManager:scheduleIn(math.max(math.floor(self:remaining())%60, 0.001), self.update_status_bars, self)
```

Comment:
What/when does this do what ? (Just having a hard time understanding it, easier to ask :))